src/dotnet/commands/dotnet-build/ProjectGraphNode.cs

```diff
@@ -0,0 +1,26 @@
+using Microsoft.DotNet.ProjectModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Tools.Build
+{
+    public class ProjectGraphNode
+    {
+        Task<ProjectContext> _projectContextCreator;
+
+        public ProjectGraphNode(Task<ProjectContext> projectContext, IEnumerable<ProjectGraphNode> dependencies, bool isRoot = false)
+        {
+            _projectContextCreator = projectContext;
+            Dependencies = dependencies.ToList();
+            IsRoot = isRoot;
+        }
+
+        public ProjectContext ProjectContext { get { return _projectContextCreator.GetAwaiter().GetResult(); } }
```

Comment:
Hrm... still makes me a little queasy :). We're going to flow async up eventually, right?